pkg/workloads/cortex/lib/client/tensorflow.py

```diff
@@ -24,47 +24,73 @@
 
 from cortex.lib.exceptions import UserRuntimeException, UserException, CortexException
 from cortex.lib.log import cx_logger
+from cortex.lib.type.model import Model, get_signature_keys, get_model_indexes, get_model_names
 
 
 class TensorFlowClient:
-    def __init__(self, tf_serving_url, signature_key):
+    def __init__(self, tf_serving_url, models):
         """Setup gRPC connection to TensorFlow Serving container.
 
         Args:
             tf_serving_url (string): Localhost URL to TF Serving container.
-            signature_key (string): The key to a signature in SignatureDefs in the model being served.
+            models        ([Model]): List of models deployed with TF serving container.
         """
+        self._tf_serving_url = tf_serving_url
+        self._models = models
+        self._model_names = get_model_names(models)
+        self._model_indexes = get_model_indexes(models)
+
         channel = grpc.insecure_channel(tf_serving_url)
         self._stub = prediction_service_pb2_grpc.PredictionServiceStub(channel)
-        self._signature = get_signature_def(self._stub)
-        parsed_signature_key, parsed_signature = extract_signature(self._signature, signature_key)
-        self._signature_key = parsed_signature_key
-        self._input_signature = parsed_signature
-        self._tf_serving_url = tf_serving_url
 
-    def predict(self, model_input):
+        # in the same order as self._models
+        self._signatures = get_signature_defs(self._stub, models)
```

Comment:
I think we can convert this to map, and then we can remove `Model.get_model_indexes`